### DISCUSSION TITLE
Reach sampler seed limit

### DISCUSSION THREAD

**Discussion by zichenwang01** [mitsuba-renderer/mitsuba3#1014] [Q&A]
Hi,
I was calling `sampler.seed(seed, wavefront_size)` and due to a too large `wavefront_size`, ran into the following error
```
File "/home/zw336/IR/bp/integrators/base.py", line 24, in prepare_sampler
    sampler.seed(seed, wavefront_size)
TypeError: seed(): incompatible function arguments. The following argument types are supported:
    1. (self: mitsuba.cuda_ad_rgb.Sampler, seed: int, wavefront_size: int = 4294967295) -> None

Invoked with: IndependentSampler[
  base_seed = 0
  sample_count = 16384
  samples_per_wavefront = 16384
  wavefront_size = 0
], 0, 4294967296
```

I was rendering a 512*512 image with `spp=2**14`. Is there any way that I could make `spp` larger? Thanks!

**Comment by njroussel** [comment]
Hi @zichenwang01 


Unfortunately, no.

OptiX allows launches up to `2**30` samples, and Dr.Jit can do some automatic splitting of larger kernels to accommodate up to `2**32` samples. Anything above that should be handled in multiple passes. For example, Mitsuba will render in multiple passes by default if the kernel is too wide: https://github.com/mitsuba-renderer/mitsuba3/blob/master/src/render/integrator.cpp#L247-L265
